### PR TITLE
relaxed binding snippet to get feedback

### PR DESCRIPTION
**Pull request by malaffoon**
@ftdholbrook, can you re-review, i finished fleshing it out. Instead of using mixin classes i put the JSON annotations directly in place; the only purpose of the classes is for marshaling so i figure that's okay? 

----
FYI Dave, i did figure out how to use the Spring relaxed binder ... problem is i don't see a hook for getting it into the YAML object mapper. Thus, this approach.

### CODE REVIEW

File: admin-service/build.gradle [hunk 1]
```diff
@@ -26,8 +26,9 @@ dependencies {
     compile 'org.springframework.cloud:spring-cloud-stream'
     compile 'org.springframework:spring-context-support'
     compile 'org.xmlunit:xmlunit-core'
+    compile 'com.fasterxml.jackson.core:jackson-annotations'
     compile 'com.fasterxml.jackson.datatype:jackson-datatype-jsr310'
-    compile 'com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.8.11'
+    compile 'com.fasterxml.jackson.dataformat:jackson-dataformat-yaml'
```
Comment (malaffoon) [inline admin-service/build.gradle]: As part of this, i had to upgrade to jackson 2.9; i updated common to have 2.9, and cleaned up the version management in reporting.

File: admin-service/src/test/java/org/opentestsystem/rdw/admin/multitenant/configserver/ObjectMapperTest.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    private ObjectMapper ymlMapper;
+
+    @Before
+    public void createMapper() {
+        ymlMapper = new ObjectMapper(new YAMLFactory()
+                .disable(YAMLGenerator.Feature.WRITE_DOC_START_MARKER)
+        );
+        ymlMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        ymlMapper.findAndRegisterModules();
+        ymlMapper.addMixIn(ArchivePropertiesTenant.class, ArchivePropertiesTenantMixin.class);
```
Comment (malaffoon) [inline admin-service/src/test/java/org/opentestsystem/rdw/admin/multitenant/configserver/ObjectMapperTest.java]: We'd add this to the object mapper used by the DefaultConfigServerClient

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/configserver/ArchivePropertiesTenantMixin.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+import com.fasterxml.jackson.annotation.JsonAlias;
+
+/**
+ * The YAML ObjectMapper doesn't have a good hook for resolving bean setters.
+ * But it would be really nice to support Spring's relaxed binding when reading properties.
+ * This mixin effectively does that: it will read properties using relaxed binding.
+ * Note that it always writes properties using the formal declaration which is camelCase.
+ */
+abstract class ArchivePropertiesTenantMixin {
```
Comment (malaffoon) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/configserver/ArchivePropertiesTenantMixin.java]: i've just done one to demonstrate the technique, we'd have to do mixins for the other config POJOs

File: admin-service/build.gradle (hunk 1 above)
Comment (ftdholbrook) [reply admin-service/build.gradle]: Cool, I was looking into that also.

File: admin-service/src/test/java/org/opentestsystem/rdw/admin/multitenant/configserver/ObjectMapperTest.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+public class ObjectMapperTest {
+
+    private ObjectMapper ymlMapper;
+
+    @Before
+    public void createMapper() {
+        ymlMapper = new ObjectMapper(new YAMLFactory()
+                .disable(YAMLGenerator.Feature.WRITE_DOC_START_MARKER)
+        );
+        ymlMapper.findAndRegisterModules();
```
Comment (malaffoon) [inline admin-service/src/test/java/org/opentestsystem/rdw/admin/multitenant/configserver/ObjectMapperTest.java]: no more special configuration for object mapper

File: build.gradle [hunk 5]
```diff
@@ -21,7 +21,7 @@ plugins {
 apply plugin: 'io.spring.dependency-management'
 
 // Set version dependencies. This allows the cli to force version, e.g. `-Pcommon=1.1.0-SNAPSHOT`
-String rdwCommonVersion = project.hasProperty("common") ? project.property("common") : "1.4.0-175"
+String rdwCommonVersion = project.hasProperty("common") ? project.property("common") : "1.4.0-SNAPSHOT"
```
Comment (malaffoon) [inline build.gradle]: will change once common build is ready

File: common/src/main/java/org/opentestsystem/rdw/reporting/common/configuration/AggregateReportingPropertiesTenant.java [hunk 6]
```diff
@@ -1,12 +1,18 @@
 package org.opentestsystem.rdw.reporting.common.configuration;
 
+import com.fasterxml.jackson.annotation.JsonAlias;
 import com.fasterxml.jackson.annotation.JsonFormat;
-import com.fasterxml.jackson.annotation.JsonInclude;
 import com.google.common.base.Objects;
 
 import java.util.Set;
 
-@JsonInclude(JsonInclude.Include.ALWAYS)
```
Comment (malaffoon) [inline common/src/main/java/org/opentestsystem/rdw/reporting/common/configuration/AggregateReportingPropertiesTenant.java]: This is the default behavior now. I'm hopeful there was a test in place that proves that.

Comment (ftdholbrook) [reply common/src/main/java/org/opentestsystem/rdw/reporting/common/configuration/AggregateReportingPropertiesTenant.java]: I think you need that in place for the spring controllers still, it is serialized de-serialized in multiple places by different object mappers.

Comment (ftdholbrook) [reply common/src/main/java/org/opentestsystem/rdw/reporting/common/configuration/AggregateReportingPropertiesTenant.java]:
```  jackson:
    default-property-inclusion: non_null```
spring property changes controller behavior

Comment (malaffoon) [reply common/src/main/java/org/opentestsystem/rdw/reporting/common/configuration/AggregateReportingPropertiesTenant.java]: oh yeah, good point ... will revert

**Pull request merged**